src/Node.php

```diff
@@ -347,22 +347,21 @@ public function getNodeKindName() : string {
      */
     public function getEndPosition() {
         // TODO test invariant - start of next node is end of previous node
-        if (isset($this->parent)) {
-            $parent = $this->parent;
-            $siblings = $parent->getChildNodes();
-            foreach ($siblings as $idx=>$nextSibling) {
-                if (spl_object_hash($nextSibling) === spl_object_hash($this)) {
-                    $siblings->next();
-                    $nextSibling = $siblings->current();
-                    return $nextSibling !== null
-                        ? $nextSibling->getFullStart()
-                        : $this->getRoot()->endOfFileToken->fullStart;
-                }
+        if (!isset($this->parent)) {
+            throw new \Exception("Unhandled node: " );
+        }
+
+        $parent = $this->parent;
+        $siblings = $parent->getChildNodes();
+        foreach ($siblings as $idx=>$nextSibling) {
+            if (spl_object_hash($nextSibling) !== spl_object_hash($this)) {
+                continue;
             }
-        } elseif ($this instanceof Script) {
-            return $this->endOfFileToken->getEndPosition();
```

Comment:
Oh man, you're right! It should actually be SourceFileNode now. I must have missed this when I converged the node kind constants and the class names (in preparation for removing node kind constants altogether.)